### PR TITLE
Define and activate AWS 11.3.2 release

### PR DESCRIPTION
**Pull request by sslavic**
Towards https://github.com/giantswarm/giantswarm/issues/11304

### CODE REVIEW

**Comment by cokiengchiara** [comment]
And @jgsqware said in Slack it's ok w him.

**Comment by sslavic** [comment]
Tested on giraffe:
- 11.3.2 cluster creation
   - this uncovered a bug in aws-cni-restarter, v1.0.2 was not published to shanghai registry, it was likely not tested on giraffe/China before, earlier aws-cni-restarter and platform releases using them could still be affected; I've manually retagged aws-cni-restarter 1.0.2 to v1.0.2; @paurosello please fix the other ones and future releases
- upgrade from 11.3.1 to 11.3.2 cluster
   - this uncovered external-dns issue present even in latest just released upstream external-dns 0.7.2 - even with prefer CNAM flag on, when upgrading cluster ingress A(+TXT) record set did not get automatically replaced with CNAME;
   - manual workaround is to, upon cluster upgrade, delete A + TXT record set for cluster's hosted zone; then external-dns will recreate CNAME + TXT record set

TODO:
- [ ] document in release notes this known issue and workaround, advising customer to have their SEs apply the workaround for them
- [ ] report prefer CNAME cluster upgrades issue upstream, if not fix it as well, upgrade external-dns-app eventually once fix is available

**Comment by cokiengchiara** [comment]
> * 11.3.2 cluster creation
>   
>   * this uncovered a bug in aws-cni-restarter, v1.0.2 was not published to shanghai registry, it was likely not tested on giraffe/China before, earlier aws-cni-restarter and platform releases using them could still be affected; I've manually retagged aws-cni-restarter 1.0.2 to v1.0.2; @paurosello please fix the other ones and future releases

@sslavic customer doesn't need to do anything here and we don't need it in release notes, right?

**Comment by cokiengchiara** [comment]
> * upgrade from 11.3.1 to 11.3.2 cluster
>   
>   * this uncovered external-dns issue present even in latest just released upstream external-dns 0.7.2 - even with prefer CNAM flag on, when upgrading cluster ingress A(+TXT) record set did not get automatically replaced with CNAME;
>   * manual workaround is to, upon cluster upgrade, delete A + TXT record set for cluster's hosted zone; then external-dns will recreate CNAME + TXT record set

Is this issue present only when upgrading from 11.3.1 to 11.3.2? Or does it also happen when creating a new 11.3.2 cluster?

**Comment by sslavic** [comment]
Added a note on known external-dns issue and manual workaround, @cokiengchiara and @jgsqware please have a look
https://github.com/giantswarm/releases/pull/304/files#diff-7a9dda5a8d8977eb47fc14f4b658b339

**Comment by sslavic** [comment]
> Is this issue present only when upgrading from 11.3.1 to 11.3.2? Or does it also happen when creating a new 11.3.2 cluster?

@cokiengchiara Vaclav's external-dns change in 1.2.1 external-dns-app and 11.3.2 platform release works OK for new cluster creation.

**Comment by sslavic** [comment]
> customer doesn't need to do anything here and we don't need it in release notes, right?
@cokiengchiara yes, I applied manual mitigation (distributed the image to China Shanghai registry) for 11.3.2 and other releases using aws-cni-restart v1.0.2

Older releases using older aws-cni-restart versions may still be affected; that's why I pinged Pau; @paurosello would you prefer me to create a ticket for you to track analyzing and fixing this, or will you create a ticket yourself?

**Comment by sslavic** [comment]
@cokiengchiara probably you need to craft a new announcement, at least to mention there's a related known external-dns issue affecting cluster upgrades, direct customers to read full release notes for details.

Please LMK when:
- [ ] you're satisfied with the updated release notes, and 
- [ ] announcement is ready, so release can be shipped to customer.

File: release-notes/aws/v11.3.2.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+## :zap: Giant Swarm Release v11.3.2 for AWS :zap:
+
+**If you are upgrading from 11.3.1, upgrading to this release will not roll your nodes. It will only update the apps.**
+
+This release fixes an external-dns issue affecting clusters in AWS China. We added an option for the app to use CNAME records instead of ALIAS, in order for DNS resolution to work in China.
+
+Note though that the fix works for new cluster creation only. For existing cluster upgrades there's known upstream issue in external-dns - existing ingress A+TXT record sets are not being replaced with CNAME+TXT record sets even when external-dns is configured with CNAMEs as preferred.
```
Comment (cokiengchiara) [inline release-notes/aws/v11.3.2.md:7]:
```suggestion
Note though that the fix works for new cluster creation only. For cluster upgrades, there is a known upstream external-dns issue: Existing ingress A+TXT record sets do not get replaced with CNAME+TXT record sets even when external-dns is configured with CNAMEs as preferred.
```

File: release-notes/aws/v11.3.2.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+We're working with external-dns community to provide the automatic working solution for cluster upgrades too. Until then, if you're upgrading cluster from aws platform release in [11.0.1 - 11.3.1] range, a manual workaround has to be applied - make sure to delete ingress A+TXT record sets, external-dns will then automatically create CNAME+TXT record sets. It is recommended to delegate applying this workaround to your SEs.
```
Comment (cokiengchiara) [inline release-notes/aws/v11.3.2.md:9]:
```suggestion
We're working with external-dns community to fix the issue for cluster upgrades. Until then, if you're upgrading a cluster in the AWS 11.0.1 to 11.3.1 range, a manual workaround has to be applied: Delete the ingress A+TXT record sets. external-dns will then automatically create CNAME+TXT record sets. **It is recommended to delegate this to your SE.**
```

File: release-notes/aws/v11.3.2.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+## :zap: Giant Swarm Release v11.3.2 for AWS :zap:
+
+**If you are upgrading from 11.3.1, upgrading to this release will not roll your nodes. It will only update the apps.**
+
+This release fixes an external-dns issue affecting clusters in AWS China. We added an option for the app to use CNAME records instead of ALIAS, in order for DNS resolution to work in China.
+
+Note though that the fix works for new cluster creation only. For cluster upgrades, there is a known upstream external-dns issue: Existing ingress A+TXT record sets do not get replaced with CNAME+TXT record sets even when external-dns is configured with CNAMEs as preferred.
```
Comment (cokiengchiara) [inline release-notes/aws/v11.3.2.md:7]:
```suggestion
Note that the fix works for new cluster creation only. Upgrades require a manual workaround. We're working with the upstream external-dns community to fix it. In the meantime, we recommend delegating cluster upgrades to your SE.```

File: release-notes/aws/v11.3.2.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+We're working with external-dns community to fix the issue for cluster upgrades. Until then, if you're upgrading a cluster in the AWS 11.0.1 to 11.3.1 range, a manual workaround has to be applied: Delete the ingress A+TXT record sets. external-dns will then automatically create CNAME+TXT record sets. **It is recommended to delegate this to your SE.**
```
Comment (cokiengchiara) [inline release-notes/aws/v11.3.2.md:9]:
```suggestion
**Note for SEs:** When upgrading a cluster, existing ingress A+TXT record sets do not get replaced with CNAME+TXT record sets even when external-dns is configured with CNAMEs as preferred. After upgrading, delete the ingress A+TXT record sets. external-dns will then automatically create CNAME+TXT record sets.```

**Comment by sslavic** [comment]
@giantswarm/chapter-se if you're supporting customers with clusters on AWS in China, please be aware of the known issue with external-dns and manual workaround (see [the release notes](https://github.com/giantswarm/releases/pull/304/files#diff-7a9dda5a8d8977eb47fc14f4b658b339)). You may be asked to apply the workaround as follow up to cluster upgrades, until external-dns issue is fixed.

**Comment by jgsqware** [comment]
sure. There is only Adidas for now

**Comment by sslavic** [comment]
It was just metadata, actual calico version actually used in 11.3.2 is 3.10.1. Metadata has been fixed via https://github.com/giantswarm/releases/pull/307

**Pull request merged**